locust/locustfile.py

```diff
@@ -0,0 +1,184 @@
+import string
+
+from locust import between, task
+from locust.contrib.fasthttp import FastHttpUser
+
+
+def get_asset_paths(add_cachebreaker=False) -> list:
+    """Return the JS and CSS bundle paths for the site -- at the moment we have
+    no need for code splitting, etc
+
+    While production uses django-whitenoise to serve files with a cachebreaking
+    hash, the files are also available via their unhashed filename.
+    If we want to, we can ask for them after appending a random cachebreaker
+    param as a querystring to ensure we get a fresh set every time we run the
+    load tests
+    """
+
+    if add_cachebreaker:
+        cachebreaker = f"?v={random.random()}"
+    else:
+        cachebreaker = ""
+
+    return [
+        f"/static/js/bundle.js{cachebreaker}",
+        f"/static/css/bundle.css{cachebreaker}",
+    ]
+
+
+def get_random_string(length=9):
+    return "".join(random.choice(string.ascii_lowercase) for i in range(length))
+
+
+class QuickstartUser(FastHttpUser):
+    wait_time = between(5, 9)
+
+    @task(6)
+    def view_homepage(self):
+        self.client.get("/")
+        for asset_path in get_asset_paths():
+            self.client.get(asset_path)
+
+    @task(4)
+    def view_posts_page(self):
+        self.client.get(f"/posts/")
+        for asset_path in get_asset_paths():
+            self.client.get(asset_path)
+
+    @task(2)
+    def view_posts_page_paginated(self):
+        random_page = random.randrange(1, 30)
+        path = f"/posts/?page={random_page}"
+
+        with self.client.get(path, catch_response=True) as response:
+            if response.status_code in [200, 404]:
+                response.success()
+
+        for asset_path in get_asset_paths():
+            self.client.get(asset_path)
+
+    @task(2)
+    def view_posts_page_filtered(self):
+        random_topic = random.choice(["rust", "voice", "browser-extensions"])
+        path = f"/posts/?topic={random_topic}"
+        self.client.get(path)
+        for asset_path in get_asset_paths():
+            self.client.get(asset_path)
+
+    @task(4)
+    def view_events_page(self):
+        self.client.get(f"/events/")
+        for asset_path in get_asset_paths():
+            self.client.get(asset_path)
+
+    @task(2)
+    def view_events_page_paginated(self):
+        random_page = random.randrange(1, 30)
+        path = f"/events/?page={random_page}"
+
+        with self.client.get(path, catch_response=True) as response:
+            if response.status_code in [200, 404]:
+                response.success()
+
+        for asset_path in get_asset_paths():
+            self.client.get(asset_path)
+
+    @task(2)
+    def view_events_page_filtered(self):
+        random_country_code = random.choice(["GB", "US", "NO", "NL", "DE"])
+        path = f"/events/?date=past&date=future&location={random_country_code}"
+        self.client.get(path)
+        for asset_path in get_asset_paths():
+            self.client.get(asset_path)
+
```

Comment:
Note that I haven't put in a function to try out search on a Posts or Events listing page. 

We could add this, but the results so far implied it would just run up against the rate limiter if abused, else be OK